src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -188,18 +233,46 @@ def translate_as_field_op(
     assert isinstance(domain_expr, gtir.FunCall)
 
     # parse the domain of the field operator
-    domain = dace_gtir_utils.get_domain(domain_expr)
+    domain = extract_domain(domain_expr)
 
+    # The reduction identity value is used in place of skip values when building
+    # a list of neighbor values in the unstructured domain.
+    #
+    # A reduction on neighbor values can be either expressed in local view (itir):
+    # vertices @ u⟨ Vertexₕ: [0, nvertices) ⟩
+    #      ← as_fieldop(
+    #          λ(it) → reduce(plus, 0)(neighbors(V2Eₒ, it)), u⟨ Vertexₕ: [0, nvertices) ⟩
+    #        )(edges);
+    #
+    # or in field view (gtir):
+    # vertices @ u⟨ Vertexₕ: [0, nvertices) ⟩
+    #      ← as_fieldop(λ(it) → reduce(plus, 0)(·it), u⟨ Vertexₕ: [0, nvertices) ⟩)(
+    #          as_fieldop(λ(it) → neighbors(V2Eₒ, it), u⟨ Vertexₕ: [0, nvertices) ⟩)(edges)
+    #        );
+    #
+    # In local view, the list of neighbors is (recursively) built while visiting
+    # the current expression.
+    # In field view, the list of neighbors is built as argument to the current
+    # expression. Therefore, the reduction identity value needs to be passed to
+    # the argument visitor (`reduce_identity_for_args = reduce_identity`).
+    # However, when the argument visitor hits an expression returning a neighbors
+    # list (see the second if-branch), we stop carrying the reduce identity further
+    # (`reduce_identity_for_args = None`) because it is not needed aymore;
+    # the reason being that reduce operates along a single axis, which in this case
+    # corresponds to the local dimension of the list of neighbor values.
```

Comment:
I think the whole problem is that there are different reduce identities:
- `self.reduce_identity`
- `reduce_identity_for_args`

I do not get why they are split.
Also I do not fully understand the explanation.
I think it is that if you hit the neighbor built in, then you would reduce over the list of neighbor hood values of a point, and because you are _currently_ ignoring skip values, i.e. assume that all are defined, you do not have to fuill something.
So technically speaking it is not _needed_ but not wrong to do.
If there is no reason to threat is as special case then I would not treat it as special case.
Except you want to do some prep work for the skip value case, but then you have to clearly indicate this.

Reply:
`self.reduce_identity` is defined in `LambdaToDataflow`, not here. Here `reduce_identity` is an input to the function. By visiting the node arguments with `reduce_identity_for_args = None`, their `reduce_identity` will be `None`.

Setting `reduce_identity_for_args = None` will also work in case of skip values. The `reduce_identity` value is filled in in place of skip values in the context of `neighbors` itself, not in the arguments context.

You are right that it would not be wrong to set `reduce_identity_for_args` also in case of neighbors arguments. I do it because it enables a sanity check that the sequence `reduce(V2E) -> neighbors(V2E) -> reduce(C2E) -> neighbors(C2E)` is accepted, while the sequence `reduce(V2E) -> reduce(C2E) -> neighbors(V2E) -> neighbors(C2E)` is not. The latter sequence would raise the `NotImplementedError("nested reductions not supported.")` exception.

Side note. There is a simple alternative design where I can avoid passing `reduce_identity` around. It would be to write a dummy value for skip values instead of doing array access (for `neighbors`) or executing the map operation (for `map` - I cannot safely apply the map operation in case for example of division by 0). Then, in the lowering of reduce I would put -- before the reduce library node -- another local map that fills the skip values with the reduce identity. I really like this approach (separate PR) but it will result in one extra write - replacing the dummy value with the reduce identity.